Src/FluentAssertions/Equivalency/Steps/StructuralEqualityEquivalencyStep.cs

```diff
@@ -55,21 +55,24 @@ public class StructuralEqualityEquivalencyStep : IEquivalencyStep
             IMember matchingMember = FindMatchFor(selectedMember, context.CurrentNode, comparands.Subject, options);
             if (matchingMember is not null)
             {
-                var nestedComparands = new Comparands
+                if (matchingMember.IsBrowsable || !options.ExcludeNonBrowsable)
```

Comment:
If we swapped the left and right hand sides of `||` and made `IsBrowsable` lazy, we would only evaluate the `GetCustomAttribute<EditorBrowsableAttribute>()` when strictly necessary.
Could be nice with a small benchmark to see what the actual overhead of determining `IsBrowsable` is.

Reply:
Looks like it is considerably more expensive if the attribute is present, but still only a minor cost at about 1 μs (on my development laptop) per lookup. If the attribute is not present, then the same code runs in about a fifth the time.

```
|                   Method | IsBrowsable |       Mean |    Error |   StdDev |  Gen 0 | Allocated |
|------------------------- |------------ |-----------:|---------:|---------:|-------:|----------:|
| CheckIfMemberIsBrowsable |       False | 1,121.7 ns | 29.20 ns | 84.71 ns | 0.0305 |     192 B |
| CheckIfMemberIsBrowsable |        True |   241.7 ns |  4.82 ns |  9.96 ns |      - |         - |
```

Will commit the benchmark and swap the order in the `if` :-)